alica_engine/src/engine/modelmanagement/factories/TransitionConditionFactory.cpp

```diff
@@ -0,0 +1,26 @@
+#include "engine/modelmanagement/factories/TransitionConditionFactory.h"
+
+#include "engine/model/TransitionCondition.h"
+#include "engine/model/TransitionConditionRepository.h"
+#include "engine/modelmanagement/Strings.h"
+#include "engine/modelmanagement/factories/BlackboardBlueprintFactory.h"
+
+#include <string>
+
+namespace alica
+{
+TransitionCondition* TransitionConditionFactory::create(const YAML::Node& conditionNode, TransitionConditionRepository* conditionRepository)
```

Comment:
We need the conditionRepository because ModelManager::parseFile requires us to return a single AlicaElement. Because all conditions are stored in a single file and we cant return a vector of conditions, we store the conditions in the conditionsRepository.